File: packages/code-storage-typescript/src/jwt_claims.ts
import type { RefPolicies } from './types';

/** Encode per-ref policy rules for the JWT `refs` claim (array of [pattern, ops] tuples). */
export function encodeRefsClaim(refs: RefPolicies): [string, string[]][] {
  return refs.map(({ pattern, ops }) => [pattern, ops ?? []]);
}

File: packages/code-storage-go/jwt_claims.go
package storage

// encodeRefsClaim builds the JWT `refs` claim as an array of [pattern, ops] tuples.
func encodeRefsClaim(refs RefPolicyList) []any {
	out := make([]any, len(refs))
	for i, rule := range refs {
		opList := []string(rule.Ops)
		if opList == nil {
			opList = []string{}
		}
		out[i] = []any{rule.Pattern, opList}
	}
	return out
}
